server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailGetSerializer.scala

```diff
@@ -89,50 +89,49 @@ object EmailGetSerializer {
-        removeJsNull(
-          Json.obj("partId" -> Json.toJson(part.partId),
-            "blobId" -> Json.toJson(part.blobId),
-            "headers" -> Json.toJson(part.headers),
-            "size" -> Json.toJson(part.size),
-            "name" -> Json.toJson(part.name),
-            "type" -> Json.toJson(part.`type`),
-            "charset" -> Json.toJson(part.charset),
-            "disposition" -> Json.toJson(part.disposition),
-            "cid" -> Json.toJson(part.cid),
-            "language" -> Json.toJson(part.language),
-            "location" -> Json.toJson(part.location),
-            "subParts" -> part.subParts.map(list => list.map(writes)))))
-    }
+  private def removeJsNull(obj: JsObject): JsObject =
+    JsObject(obj.fields.filter({
+      case (_, JsNull) => false
+      case _ => true
+    }))
+  def writeBodyPart(properties: Properties)(part: EmailBodyPart): JsObject =
+    properties.filter(removeJsNull(Json.obj("partId" -> Json.toJson(part.partId),
+      "blobId" -> Json.toJson(part.blobId),
+      "headers" -> Json.toJson(part.headers),
+      "size" -> Json.toJson(part.size),
+      "name" -> Json.toJson(part.name),
+      "type" -> Json.toJson(part.`type`),
+      "charset" -> Json.toJson(part.charset),
+      "disposition" -> Json.toJson(part.disposition),
+      "cid" -> Json.toJson(part.cid),
+      "language" -> Json.toJson(part.language),
+      "location" -> Json.toJson(part.location),
+      "subParts" -> part.subParts.map(list => list.map(writeBodyPart(properties))))))
+
+  private def bodyPartWritesWithPropertyFilter(properties: Properties): Writes[EmailBodyPart] = writeBodyPart(properties)
+
+  private implicit val emailMetadataWrites: OWrites[EmailMetadata] = Json.writes[EmailMetadata]
+  private implicit val emailHeadersWrites: Writes[EmailHeaders] = Json.writes[EmailHeaders]
+  private implicit val emailBodyMetadataWrites: Writes[EmailBodyMetadata] = Json.writes[EmailBodyMetadata]
+
+  private val emailFastViewWrites: OWrites[EmailFastView] = (JsPath.write[EmailMetadata] and
+    JsPath.write[EmailHeaders] and
+    JsPath.write[EmailBodyMetadata] and
+    JsPath.write[Map[String, Option[EmailHeaderValue]]]) (unlift(EmailFastView.unapply))
+  private val emailHeaderViewWrites: OWrites[EmailHeaderView] = (JsPath.write[EmailMetadata] and
+    JsPath.write[EmailHeaders] and
+    JsPath.write[Map[String, Option[EmailHeaderValue]]]) (unlift(EmailHeaderView.unapply))
+  private val emailMetadataViewWrites: OWrites[EmailMetadataView] = view => Json.toJsObject(view.metadata)
 
   private def emailWritesWithPropertyFilter(properties: Properties)(implicit partsWrites: Writes[EmailBodyPart]): Writes[EmailView] = {
-    implicit val emailMetadataWrites: OWrites[EmailMetadata] = Json.writes[EmailMetadata]
-    implicit val emailHeadersWrites: Writes[EmailHeaders] = Json.writes[EmailHeaders]
     implicit val emailBodyWrites: Writes[EmailBody] = Json.writes[EmailBody]
```

Comment:
Could you move this line out of the method in order not to do it at each serialization ?

Reply:
No it depends on the implicit partsWrites.... (which may recurse, and be filtered itself...)